syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -157,15 +160,22 @@ func (j *archiveParser) discoverMainPackage() (*pkg.Package, error) {
 		return nil, nil
 	}
 
+	// grab and assign digest for the entire archive
+	digest, err := syftFile.CalculateDigest(j.archivePath, syftFile.HashAlgoSHA1)
+	if err != nil {
+		log.Warnf("failed to parse digest for file (%s): %+v", j.archivePath, err)
+	}
+
 	return &pkg.Package{
 		Name:         selectName(manifest, j.fileInfo),
 		Version:      selectVersion(manifest, j.fileInfo),
 		Language:     pkg.Java,
 		Type:         j.fileInfo.pkgType(),
 		MetadataType: pkg.JavaMetadataType,
 		Metadata: pkg.JavaMetadata{
-			VirtualPath: j.virtualPath,
-			Manifest:    manifest,
+			VirtualPath:    j.virtualPath,
+			Manifest:       manifest,
+			ArchiveDigests: []syftFile.Digest{*digest},
```

Comment:
there isn't a nil check, so there could be a panic with `*digest`. It could be that a new `file.NewDigestOfFile(path, crypto.Hash...) ([]Digest, error)` could naturally return what you're looking for without a nil check.

Reply:
Yea good call here - I'll just refactor based on the suggestion above to expose what's in the digest cataloger already. I can return by value here as well only if an error is not returned so we don't have to assert on the non error return value